app/controllers/ips/remove_controller.rb

```diff
@@ -0,0 +1,5 @@
+class Ips::RemoveController < ApplicationController
```

Comment:
I would put this on the `IpsController`.

The resource we're dealing with here is `ip`, and all this action does is `show` it - even though we know you can only delete it from this page. I don't think the extra controller makes it clearer.

Reply:
Actually this is intentional, and kind of the nexus of the whole thing. Basically we need a 2-step process to delete an IP. Step one is clicking remove, step two is confirming the remove. We can't have a pop-up or JS, so there needs to be a page refresh, which means there needs to be two controller endpoints handling this. (It could be done as two steps via the same endpoint but thats really ugly).

What this does is keep all our routes and controllers RESTful. It does this by creating the concept of an "IP removal". Step one takes you to an "Ip removal show page", or in other words "shows you the ip you want to remove", and then step two does the actual delete on the IP.

If this was all done on the `IpsController` it would not be RESTful as the actions would be:
- `ips#display_removal` (renders a page or partial that shows a delete button)(not RESTful)
- `ips#destroy` (deletes the ip)

Instead we have more, smaller, RESTful controllers:
- `ips/remove#show`  (renders a page or partial that shows a delete button)
- `ips#destroy`  (deletes the ip)
 
This pattern of many smaller RESTful controllers is a Rails pattern to prevent controllers from getting out of hand. http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/

Reply:
I understand (and agree with) the desire to make this RESTful, and the linked blog post. I don't think the creation of an extra resource (the abstract "IP deletion request") is the right choice in this case, though.

If this was done on the IpsController, the actions could be
- `ips#show` (renders the page showing the IP details and a delete button)
- `ips#destroy` (deletes the ip)

Maybe you're not happy with this being a `show` page, because it mainly emphasises this one use case - deleting the IP?

I think having a separate controller is less clear than the `show` page being mainly about deletion.

Reply:
Yeah this page just doesn't feel like an IP show page. I agree having the `ips#show` display this page is simpler, but then the routes and url don't really make sense, eg:

`ips#show`:
- `ip_path(ip)`
- `wifi.service.gov.uk/ips/2345`

`ips/remove#show`:
- `ip_remove_path(ip)`
- `wifi.service.gov.uk/ips/2345/remove`

Reply:
I'm happy with either of those routes, but I'm also happy if you want to separate this out.